tasks/request_files/request_files.py

```diff
@@ -203,62 +228,127 @@ def inner_task(
+                #       to failure instead of pending with a completion_time
+                #       and error_message value.
+                LOGGER.warn(f"{file_key} does not exist in S3 bucket")
+
+        # Create a copy of the granule and add file information in the proper
+        # format
         copied_granule = granule.copy()
         copied_granule[GRANULE_RECOVER_FILES_KEY] = files
 
-    # todo: Looks like this line is why multiple granules are not supported.  ORCA-202
-    # todo: Using the default value {} for copied_granule will cause this function to raise errors every time.  ORCA-202
-    process_granule(
-        s3,
-        copied_granule,
-        glacier_bucket,
-        restore_expire_days,
-        max_retries,
-        retry_sleep_secs,
-        retrieval_type,
-        event[EVENT_INPUT_KEY][INPUT_JOB_ID_KEY],
-        db_queue_url,
-    )
+        # Send initial job and status information to the database queues
+        # post to DB-queue. Retry using exponential delay if it fails
+        LOGGER.debug("Sending initial job status information to DB QUEUE.")
+        job_id = event[EVENT_INPUT_KEY][INPUT_JOB_ID_KEY]
+        granule_id = granule[GRANULE_GRANULE_ID_KEY]
+
+        for retry in range(max_retries + 1):
+            try:
+                shared_recovery.create_status_for_job(
+                    job_id, granule_id, glacier_bucket, files, db_queue_url
+                )
+                break
+            except Exception as ex:
+                # todo: Workaround for CumulusLogger bug with dictionaries
+                #       this will need updating when a new CumulusLogger is
+                #       released with bug fix.
+                msg = f"Ran into error posting to SQS {retry+1} time(s) with exception"
+                msg = msg + " {ex}"
+                LOGGER.error(msg, ex=str(ex))
+                # todo: Use backoff code. ORCA-201
+                time.sleep(retry_sleep_secs)
+                continue
+        else:
+            message = f"Unable to send message to QUEUE {db_queue_url}"
+            LOGGER.critical(message, exec_info=True)
+            raise Exception(message)
+
+        # todo: Looks like this line is why multiple granules are not supported.
+        #       will need to update see ORCA-202
+        # todo: Using the default value {} for copied_granule will cause this
+        #       function to raise errors every time.  ORCA-202
+        #
+        # Process the granules by restoring them from glacier and updating the
+        # database with any failure information.
+        process_granule(
+            s3,
+            copied_granule,
+            glacier_bucket,
+            restore_expire_days,
+            max_retries,
+            retry_sleep_secs,
+            retrieval_type,
+            job_id,
+            db_queue_url,
+        )
+
+        # If no errors add to copied_granules array for output.
```

Comment:
I might remove the "If no errors" since that isn't really a factor beyond as much as it is for any line.

Reply:
Did you mean to have a try/except around process_granule?